stix_shifter_modules/aws_athena/stix_transmission/post_query_connector_error_handling.py

```diff
@@ -0,0 +1,112 @@
+    
+from venv import logger
+import regex as re
+import time
+from stix_shifter_modules.aws_athena.stix_transmission import status_connector
+
+class PostQueryConnectorErrorHandler():
+    async def check_status_for_missing_column(client, search_id, query):
+        """Creates a status check loop to see if the query fails with a column doesn't exist exception. If it does, return the query with the offending column removed.
+            If it does not, return with "CONNECTOR_FACTORY_SUCCESS"
+
+        Args:
+            client (RestApiClientAsync) 
+            search_id (String): For each query sent to Athena, a job is created that runs in till it finishes. This ID is used to find the job and access it's results.
+            query (String): The query that will be modified if a missing column error occurs.
+
+        Returns:
+            String: Returns either a modified query, or a special key that can be used to know that the query was successful and to stop.
+        """        
+        status = status_connector.StatusConnector(client)
+        column_to_delete = ""
+        
+        #Wait ten seconds if the status is RUNNING. Exits early if the status is not RUNNING.
+        for i in range(10):
+            time.sleep(1)
+            status_response = await status.create_status_connection(search_id)
+            #Checks if there is a message that can be read and if that message matches the column not found message.
+            if(status_response != None and "message" in  status_response):
+                match = re.search(f"Column '(.*)' cannot be resolved", status_response["message"])
+                if(match):
+                    #If there is a match, return the column name.
+                    column_to_delete = match.group(1)
+                    break
+            elif(status_response != None and "status" in status_response and status_response["status"] != "RUNNING"):
+                #If there is no match and the status is not running, than stop trying and exit with the success message.
+                break
+                
+        if(column_to_delete != ""):
+            return PostQueryConnectorErrorHandler._remove_invalid_column_table(column_to_delete, query)
+        else:
+            #May not always be successful, just that no column error is occurring.
+            return "CONNECTOR_FACTORY_SUCCESS"
```

Comment:
i would return a boolean True instead of this odd looking string. you can change the function declaration like this:

`async def check_status_for_missing_column(client, search_id, query) -> None:`